arenadata/Dockerfile

```diff
@@ -1,5 +1,17 @@
 FROM rockylinux:8.8 as base
 
+# Allow usage of custom compilers, however, gcc is the default
+ARG CC=gcc
+ARG CXX=g++
+ARG SKIP_UNITTESTS=false
+ARG USE_CLANG_ANALYZER=false
+
+RUN \
+    if [ "$USE_CLANG_ANALYZER" = true ]; then \
+        export CFLAGS="-fsyntax-only --analyze"; \
+        export CXXFLAGS="-fsyntax-only --analyze"; \
```

Comment:
I think these exports only affect the current run, but not the next ones.

Reply:
What do you think about placing it in compile_gpdb.bash?

Reply:
> What do you think about placing it in compile_gpdb.bash?

Why not

Reply:
Either way is fine, I mean if it won't work via dockerfile then we can move it somewhere in scripts

Reply:
Adding -fsyntax-only breaks configure script as it tries to compile a test program, but clang doesn't yield anything. I don't think that adding them is worth the work it requires

Reply:
hm...
![screen](https://github.com/user-attachments/assets/8c3fbf96-57fc-4d2f-accd-28392fa24334)